external-crates/move/crates/move-package/src/compilation/build_plan.rs

```diff
@@ -130,12 +136,20 @@ impl BuildPlan {
     }
 
     /// Compilation process does not exit even if warnings/failures are encountered
-    pub fn compile_no_exit<W: Write>(&self, writer: &mut W) -> Result<CompiledPackage> {
+    pub fn compile_no_exit<W: Write>(
+        &self,
+        writer: &mut W,
+        modify_compiler: impl FnOnce(Compiler) -> Compiler,
+    ) -> Result<CompiledPackage> {
         self.compile_with_driver(writer, |compiler| {
-            let (files, units_res) = compiler.build()?;
+            let (files, units_res) = modify_compiler(compiler).build()?;
             match units_res {
                 Ok((units, warning_diags)) => {
-                    report_warnings(&files, warning_diags);
+                    let diags_buf =
+                        report_diagnostics_to_buffer_with_env_color(&files, warning_diags);
+                    if let Err(err) = std::io::stdout().write_all(&diags_buf) {
+                        anyhow::bail!("Cannot output compiler diagnostics: {}", err);
+                    }
```

Comment:
Calling `std::io:stdout` in the middle of this code seems a little out of place versus the helper.

Reply:
I agree, but this was closer to what the old code did. I'll see what I can do though